kai/import_from_hub.py

```diff
@@ -0,0 +1,338 @@
+    if validated_analyses:
+        return validated_analyses[0].id
+
+    return last_analysis
+
+
+def get_data_from_api(url: str, params=None, timeout: int = 60, verify: bool = True):
+    if not params:
+        params = {}
+    response = requests.get(url, params=params, timeout=timeout, verify=verify)
+    response.raise_for_status()
+    return response.json()
+
+
+def process_analyses(
+    analyses: List[Analysis],
+    base_url: str,
+    application_dir: str,
+    request_timeout: int = 60,
+    request_verify: bool = True,
+) -> List[Tuple[Application, Report]]:
+
+    reports: List[Tuple[Application, Report]] = []
+    for analysis in analyses:
+        KAI_LOG.info(
+            f"Processing analysis {analysis.id} for application {analysis.application.id}"
+        )
+        application = parse_application_data(
+            get_data_from_api(
+                f"{base_url}/hub/applications/{analysis.application.id}",
+                timeout=request_timeout,
+                verify=request_verify,
+            ),
+            application_dir,
+        )
+        application.current_commit = analysis.commit
+        report_data = {}
+        issues_url = f"{base_url}/hub/analyses/{analysis.id}/issues"
+        issues = [
+            Issue(**item)
+            for item in get_data_from_api(
+                issues_url, timeout=request_timeout, verify=request_verify
+            )
+        ]
+        for issue in issues:
+            KAI_LOG.info(
+                f"Processing issue {issue.id} with effort {issue.effort} on ruleset {issue.ruleset} (commit: {analysis.commit})"
+            )
+            key = issue.ruleset
+            if key not in report_data:
+                report_data[key] = {"description": issue.description, "violations": {}}
+            for incident in issue.incidents:
+                incident.file = incident.file.removeprefix(
+                    f"/addon/source/{application.application_name}/"
+                )
+                incident.uri = incident.uri.removeprefix(
+                    f"/addon/source/{application.application_name}/"
+                )
+                KAI_LOG.debug(f"{incident.variables=}")
+            report_data[key]["violations"][issue.rule] = {
+                "category": issue.category,
+                "description": issue.description,
+                "effort": issue.effort,
+                "incidents": issue.incidents,
+            }
+        if report_data:
+            reports.append((application, Report.load_report_from_object(report_data)))
+    return reports
+
+
+def clone_repo_at_commit(repo_url, branch, commit, destination_folder):
+    try:
+        subprocess.run(
+            ["git", "clone", "-b", branch, repo_url, destination_folder], check=True
+        )
+    except subprocess.CalledProcessError as e:
+        KAI_LOG.error(f"An error occurred: {e}")
+
+    try:
+        subprocess.run(["git", "checkout", commit], cwd=destination_folder, check=True)
```

Comment:
Consider adding the `trunk-ignore` comments if these are spurious

Reply:
yeah a bunch of trunk comments still need to be added